Demo-Swift/Demo-Swift/MHAudioBufferPlayer/MHAudioBufferPlayer.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  MHAudioBufferPlayer.swift
+//  Demo-Swift
+//
+//  Created by Kelvin Lau on 2018-01-11.
+//  Copyright © 2018 Kelvin Lau. All rights reserved.
+//
+
+import AudioToolbox
+import AVFoundation
+
+typealias MHAudioBufferPlayerBlock = (_ buffer: AudioQueueBufferRef, _ audioFormat: AudioStreamBasicDescription) -> Void
+let NumberOfAudioDataBuffers = 3
+
+func InterruptionListenerCallback(inUserData: UnsafeRawPointer, interruptionState: UInt32) {
+  let player = Unmanaged<MHAudioBufferPlayer>.fromOpaque(inUserData).takeUnretainedValue()
+  if interruptionState == kAudioSessionBeginInterruption {
+    player.tearDownAudio()
+  } else if interruptionState == kAudioSessionEndInterruption {
+    player.setUpAudio()
+    player.start()
+  }
+}
+
+func PlayCallback(_ inUserData: UnsafeMutableRawPointer!, _ inAudioQueue: AudioQueueRef, _ inBuffer: AudioQueueBufferRef) {
+  let player = Unmanaged<MHAudioBufferPlayer>.fromOpaque(inUserData).takeUnretainedValue()
+  if player.playing && player.block != nil {
+    player.block!(inBuffer, player.audioFormat)
+    AudioQueueEnqueueBuffer(inAudioQueue, inBuffer, 0, nil)
+  }
+}
+
+class MHAudioBufferPlayer {
+  
+  var block: MHAudioBufferPlayerBlock?
+  private(set) var playing: Bool
+  var gain: Float32
+  var audioFormat: AudioStreamBasicDescription
+  
+  var _playQueue: AudioQueueRef?
+  var _playQueueBuffers: [AudioQueueBufferRef?] = .init(repeating: nil, count: NumberOfAudioDataBuffers)
```

Comment:
The original Objective-C looks like this:

```
AudioQueueBufferRef _playQueueBuffers[NumberOfAudioDataBuffers];
```

Is the Swift statement an accurate representation of that Objective-C statement?